veggieseasons/lib/main.dart

```diff
@@ -17,19 +17,73 @@ void main() {
   ]);
 
   runApp(
-    MultiProvider(
+    RootRestorationScope(
+      restorationId: 'root',
+      child: VeggieApp(),
+    ),
+  );
+}
+
+class VeggieApp extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _VeggieAppState();
+}
+
+class _VeggieAppState extends State<VeggieApp> with RestorationMixin {
+  final _RestorableAppState _appState = _RestorableAppState();
+
+  @override
+  String get restorationId => 'wrapper';
+
+  @override
+  void restoreState(RestorationBucket oldBucket, bool initialRestore) {
+    registerForRestoration(_appState, 'state');
+  }
+
+  @override
+  void dispose() {
+    _appState.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiProvider(
       providers: [
-        ChangeNotifierProvider(
-          create: (_) => AppState(),
+        ChangeNotifierProvider.value(
+          value: _appState.value,
         ),
         ChangeNotifierProvider(
           create: (_) => Preferences()..load(),
         ),
       ],
       child: CupertinoApp(
         debugShowCheckedModeBanner: false,
-        home: HomeScreen(),
+        home: HomeScreen(restorationId: 'home'),
+        restorationScopeId: 'app',
       ),
-    ),
-  );
+    );
+  }
+}
+
+class _RestorableAppState extends RestorableListenable<AppState> {
```

Comment:
Is it possible to merge this class with AppState itself, or is the better pattern to make a restorable wrapper as done here? I ask because between VeggieAppState, _RestorableAppState, and AppState, there's a whole lot of state now. 😄

Reply:
Also, when https://github.com/flutter/samples/issues/573 is fixed, the _RestorableAppState can be removed because after the bug is fixed, the state of AppState would already be persisted in shared_preferences.

Reply:
> You have one class that represents the actual value and another wrapper class that is in charge of serializing/deserializing that thing.

Got it. That makes sense. We (meaning DevRel) will want to remember that as a recommended practice when talking over state restoration.